pkg/psmdb/container.go

```diff
@@ -192,20 +192,13 @@ func containerArgs(ctx context.Context, cr *api.PerconaServerMongoDB, replset *a
 		args = append(args, "--sslAllowInvalidCertificates")
 	}
 
-	if cr.TLSEnabled() {
-		if cr.Spec.TLS.Mode == api.TLSModeAllow {
-			args = append(args,
-				"--clusterAuthMode=keyFile",
-				"--keyFile="+mongodSecretsDir+"/mongodb-key",
-			)
-		} else {
-			args = append(args, "--clusterAuthMode=x509")
-		}
-	} else if cr.UnsafeTLSDisabled() {
+	if (cr.TLSEnabled() && cr.Spec.TLS.Mode == api.TLSModeAllow) || cr.UnsafeTLSDisabled() || cr.Spec.Secrets.InternalKey != "" {
```

Comment:
`cr.Spec.Secrets.InternalKey != "" ` this will always be true because we set it by default in https://github.com/percona/percona-server-mongodb-operator/blob/dev/K8SPSMDB-1132/pkg/apis/psmdb/v1/psmdb_defaults.go#L77-L79

this is why all tests are failing